doc/source/ray-air/key-concepts.rst

```diff
@@ -0,0 +1,76 @@
+AIR Key Concepts
+================
+
+Here, we cover the main concepts that AIR is consisted of.
+
+.. contents::
+    :local:
+
+
+Preprocessors
+-------------
+
+Preprocessors are primitives that can be used to transform input data into features.
+
+A preprocessor can be fitted during Training, and applied at runtime in both Training and Serving on data batches in the same way.
+
+Preprocessors operate on :ref:`Ray Datasets <datasets>`.
+
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __air_preprocessors_start__
+    :end-before: __air_preprocessors_end__
+
+
+Trainers
+--------
+
+Trainers are wrapper classes around third-party training frameworks like XGBoost and Pytorch. They are built to help integrate with core Ray actors (for distribution), Tune, and Datasets.
+
+See the documentation on :ref:`Trainers <air-trainer-ref>`.
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __air_trainer_start__
+    :end-before: __air_trainer_end__
+
+
+
+Trainer objects will produce a :ref:`Results <air-results-ref>` object after calling ``.fit()``.  These objects will contain training metrics as long as checkpoints to retrieve the best model.
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __air_trainer_output_start__
+    :end-before: __air_trainer_output_end__
+
+
+Tuner
+-----
+
+:ref:`Tuners <air-tuner-ref>` offer scalable hyperparameter tuning as part of :ref:`Ray Tune <tune-main>`.
+
+Tuners can work seamlessly with any Trainer but also can support arbitrary training functions.
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __air_tuner_start__
+    :end-before: __air_tuner_end__
+
+
+
+Batch Predictor
+---------------
+
+You can take a trained model and do batch inference using the BatchPredictor object.
+
+.. literalinclude:: doc_code/key_concepts.py
+    :language: python
+    :start-after: __air_batch_predictor_start__
+    :end-before: __air_batch_predictor_end__
+
+
+.. Online Inference
+.. ----------------
+
+.. TODO
```

Comment:
@simon-mo is this checked in yet? do you have an example for using modelwrappers?

Reply:
not yet. can you create a ticket and assign to me? so far @krfricke has an end to end example https://github.com/ray-project/ray/commit/76da2255d9c59d485e8cd5e942de49eb93e03011